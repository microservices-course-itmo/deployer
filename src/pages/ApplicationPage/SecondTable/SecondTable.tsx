import React from 'react'

import { IApplicationData } from '../../../types/Application'
import { EnvironmentsTable } from './EnvironmentsTable'
import { PortsTable } from './PortsTable'
import { VolumesTable } from './VolumesTable'

interface ISecondTable {
  data: IApplicationData
  type: number
  className?: string
}

export const SecondTable = ({ className, type, data }: ISecondTable) => {
  const components = [<EnvironmentsTable data={data} />, <PortsTable data={data} />, <VolumesTable data={data} />]

  return <div className={className}>{components[type]}</div>
}
