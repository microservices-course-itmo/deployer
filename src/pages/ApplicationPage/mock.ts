export const mockData = {
  templateVersion: '15',
  createdBy: 'asukhoa',
  name: 'order-service',
  volumes: ['opt/kafka'],
  description: 'service for ...',
  lastRelease: '1.0.1',
  'port-mappings': [
    { port: '8080', value: '8080' },
    { port: '8081', value: '48081' },
  ],
  env: [
    { name: 'POSTGRES_HOST', value: 'postgres' },
    { name: 'POSTGRES_PORT', value: '5432' },
  ],
  instances: [
    {
      id: '1',
      appId: 'order-service',
      templateId: '15',
      version: '1.0.1',
      containerId: '12235123512',
      dateCreated: '23412351245',
      userCreated: 'asukhoa',
      status: 'RUNNING',
      alias: 'new feature added',
      url: '127.0.0.1:48080/order-service/swagger-ui.html',
    },
    {
      id: '2',
      appId: 'order-service',
      templateId: '15',
      version: '1.0.2',
      containerId: '12235123512',
      dateCreated: '23412351245',
      userCreated: 'asukhoa',
      status: 'FAILED',
      alias: 'new feature added',
      url: '127.0.0.1:48080/order-service/swagger-ui.html',
    },
    {
      id: '3',
      appId: 'order-service',
      alias: 'new_postgres',
      templateId: '15',
      version: '1.0.3',
      containerId: '12235123512',
      dateCreated: '23412351245',
      userCreated: 'asukhoa',
      status: 'STARTING',
      url: '127.0.0.1:48080/order-service/swagger-ui.html',
    },
    {
      id: '4',
      appId: 'order-service',
      templateId: '15',
      version: '1.0.4',
      containerId: '12235123512',
      dateCreated: '23412351245',
      userCreated: 'asukhoa',
      status: 'STOPPED',
      alias: 'new feature added',
      url: '127.0.0.1:48080/order-service/swagger-ui.html',
    },
    {
      id: '5',
      appId: 'order-service',
      templateId: '15',
      version: '1.0.5',
      containerId: '12235123512',
      dateCreated: '23412351245',
      userCreated: 'asukhoa',
      status: 'REMOVED',
      alias: 'new feature added',
      url: '127.0.0.1:48080/order-service/swagger-ui.html',
    },
    {
      id: '6',
      appId: 'order-service',
      templateId: '15',
      version: '1.0.53',
      containerId: '12235123512',
      dateCreated: '23412351245',
      userCreated: 'asukhoa',
      status: 'RESTARTING',
      alias: 'new feature added',
      url: '127.0.0.1:48080/order-service/swagger-ui.html',
    },
  ],
  versions: ['2.0.0', '2.0.1', '2.0.2', '2.0.3', '2.0.4'],
  history: [
    {
      id: '1',
      date: '02/10/2020',
      user: 'User',
      log: 'created new env var',
      prevState: {
        env: [['POSTGRES_HOST', 'postgres']],
        volumes: ['otp/kafka'],
        ports: [],
      },
      currentState: {
        env: [
          ['POSTGRES_HOST', 'postgres'],
          ['POSTGRES_PORT', '5432'],
        ],
        volumes: ['otp/kafka'],
        ports: [
          ['8080', '8080'],
          ['8081', '48081'],
        ],
      },
    },
    {
      id: '2',
      date: '04/10/2020',
      user: 'User',
      log: 'forward new port',
      prevState: {
        env: [['POSTGRES_HOST', 'postgres']],
        volumes: ['otp/kafka'],
        ports: [],
      },
      currentState: {
        env: [
          ['POSTGRES_HOST', 'postgres'],
          ['POSTGRES_PORT', '5432'],
        ],
        volumes: ['otp/kafka'],
        ports: [
          ['8080', '8080'],
          ['8081', '48081'],
        ],
      },
    },
    {
      id: '3',
      date: '06/10/2020',
      user: 'User',
      log: '...',
      prevState: {
        env: [['POSTGRES_HOST', 'postgres']],
        volumes: ['otp/kafka'],
        ports: [],
      },
      currentState: {
        env: [
          ['POSTGRES_HOST', 'postgres'],
          ['POSTGRES_PORT', '5432'],
        ],
        volumes: ['otp/kafka'],
        ports: [
          ['8080', '8080'],
          ['8081', '48081'],
        ],
      },
    },
    {
      id: '4',
      date: '07/10/2020',
      user: 'User',
      log: 'created new env var',
      prevState: {
        env: [['POSTGRES_HOST', 'postgres']],
        volumes: ['otp/kafka'],
        ports: [],
      },
      currentState: {
        env: [
          ['POSTGRES_HOST', 'postgres'],
          ['POSTGRES_PORT', '5432'],
        ],
        volumes: ['otp/kafka'],
        ports: [
          ['8080', '8080'],
          ['8081', '48081'],
        ],
      },
    },
    {
      id: '5',
      date: '09/10/2020',
      user: 'User',
      log: '...',
      prevState: {
        env: [['POSTGRES_HOST', 'postgres']],
        volumes: ['otp/kafka'],
        ports: [],
      },
      currentState: {
        env: [
          ['POSTGRES_HOST', 'postgres'],
          ['POSTGRES_PORT', '5432'],
        ],
        volumes: ['otp/kafka'],
        ports: [
          ['8080', '8080'],
          ['8081', '48081'],
        ],
      },
    },
  ],
}
