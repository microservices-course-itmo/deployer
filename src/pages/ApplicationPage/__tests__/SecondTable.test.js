import React from 'react'
import { SecondTable } from '../SecondTable'
import { mockData } from '../mock'
import renderer from 'react-test-renderer'

describe('SecondTable', ()=>{

it('renders correctly', () => {
    const depcheck = renderer
        .create(<SecondTable data={mockData} type={0}/>)
        .toJSON()
    expect(depcheck).toMatchSnapshot()
});
})