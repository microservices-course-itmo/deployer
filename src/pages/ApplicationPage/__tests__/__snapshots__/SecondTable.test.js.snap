// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondTable renders correctly 1`] = `
<div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        className="MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              Value
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled"
                    disabled={true}
                    id="standard-required"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value="POSTGRES_HOST"
                  />
                </div>
              </div>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input"
                    disabled={false}
                    id="standard"
                    name="POSTGRES_HOST"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value="postgres"
                  />
                </div>
              </div>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <button
                aria-label="delete"
                className="MuiButtonBase-root MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
              </button>
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled"
                    disabled={true}
                    id="standard-required"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value="POSTGRES_PORT"
                  />
                </div>
              </div>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input"
                    disabled={false}
                    id="standard"
                    name="POSTGRES_PORT"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value="5432"
                  />
                </div>
              </div>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <button
                aria-label="delete"
                className="MuiButtonBase-root MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
              </button>
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                  data-shrink={false}
                  htmlFor="standard"
                  id="standard-label"
                >
                  name
                </label>
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input"
                    disabled={false}
                    id="standard"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                  data-shrink={false}
                  htmlFor="standard"
                  id="standard-label"
                >
                  value
                </label>
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input"
                    disabled={false}
                    id="standard"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <button
                aria-label="add"
                className="MuiButtonBase-root MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    style={
                      Object {
                        "color": "#3F51B5",
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </span>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-saveBtn-1 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Save
        </span>
      </button>
    </div>
  </div>
</div>
`;
